src/contexts/Modal.tsx

```diff
@@ -0,0 +1,73 @@
+import React, {
+  ReactElement,
+  ReactNode,
+  createContext,
+  useCallback,
+  useContext,
+  useRef,
+  useState,
+} from 'react';
+
+import Modal from '@app/components/common/Modal';
+
+export interface ModalOptions {
+  disablePaper?: boolean;
+  disableIndirectDismissal?: boolean;
+  onDismiss?: () => void;
+}
+export interface ModalCtx {
+  isOpen: boolean;
+  disclose: (content: ReactElement, options?: ModalOptions) => void;
+  dismiss: (skipDismissNotification?: boolean) => void;
+}
+
+type ModalProps = Pick<ModalOptions, 'disablePaper' | 'disableIndirectDismissal'>;
+
+export const ModalContext = createContext<ModalCtx>({} as ModalCtx);
+export const useModal = (): ModalCtx => useContext(ModalContext);
+
+const ModalProvider: React.FC = props => {
+  const [isOpen, setIsOpen] = useState<boolean>(false);
+  const [isLoading, setLoading] = useState<boolean>(false);
+  const [modalContent, setModalContent] = useState<ReactNode>(null);
+  const [modalProps, setModalProps] = useState<ModalProps>({});
+  const dismissCallback = useRef<(() => void) | null>(null);
+
+  const disclose = useCallback((content: ReactElement, modalOptions: ModalOptions = {}): void => {
+    if (content.props) {
+      const clonedContent = React.cloneElement(content, {
+        setLoadingIndicator: setLoading,
+      });
+      setModalContent(clonedContent);
+    } else {
+      setModalContent(content);
+    }
+    dismissCallback.current = modalOptions.onDismiss ?? null;
+
+    const { disablePaper, disableIndirectDismissal } = modalOptions;
+    setModalProps({ disablePaper, disableIndirectDismissal });
+    setIsOpen(true);
+  }, []);
+
+  const dismiss = useCallback((skipDismissNotification?: boolean) => {
```

Comment:
I know I wrote this but I think it's better to just get rid of that parameter, I don't really see a logical case for it now and I think it just came along from another implementation we had for this at work

Reply:
donezo https://github.com/resident51/resident51/pull/4/commits/1810cf03762f138a810a74612cddc138cf602e38